datalad/support/annexrepo.py

```diff
@@ -1001,6 +1001,51 @@ def _call_annex(self, args, files=None, jobs=None, protocol=StdOutErrCapture,
             # we don't know how to handle this, just pass it on
             raise
 
+    @staticmethod
+    def _get_non_existing_from_annex_output(output):
+        """This parses annex' output for messages about non-existing paths
+        and returns a list of such paths (as strings).
+
+        Internal helper for `_call_annex_records`.
+        """
+
+        # This should cease to exist whenever annex provides an actual API
+        # for retrieving such information;
+        # see https://git-annex.brachable.com/todo/api_for_telling_when_nonexistant_or_non_git_files_passed/
+
+        # Output largely depends on annex config annex.skipunknown
+        # (see https://github.com/datalad/datalad/pull/6510#issuecomment-1054499339)
+        # and git-annex' default of annex.skipunknown changed as of 10.20220127.
+        # However, that appears to not be true for all commands. annex-add would
+        # still report in the "git-annex: ... not found" fashion rather than
+        # "error:  ... did not match any file(s) known to git". Depends on
+        # what annex is calling internally *and* on that config. Apparently git
+        # itself isn't consistent in that regard.
+        # Therefore, account for both ways of reporting unknown paths.
+
+        unknown_paths = []
+        for line in output.splitlines():
+            if 'did not match any file(s) known to git' in line:
+                unknown_paths.append(line[17:-40])
+            elif line.startswith('git-annex:') and line.endswith(' not found'):
+                unknown_paths.append(line[11:-10])
+
+        return unknown_paths
+
+    @staticmethod
+    def _fake_json_for_non_existing(paths, cmd):
+        """Create faked JSON records for nonexisting paths provided by `paths`
+        after running `cmd`.
+
+        Internal helper for `_call_annex_records`.
+        """
+
+        return [{"command": cmd,
+                 "file": f,
+                 "note": "not found",
+                 "success": False,
+                 } for f in ensure_list(paths)]
+
```

Comment:
These two are just functions. I see no reason to put them in `annexrepo.py` -- which is one of the largest and hardest to navigate source files we have (almost 4k lines). Please put them elsewhere (and not in `datalad/utils.py`, 2.7k lines). Thanks

Reply:
Not saying that this is the best place, but `annexjson2result()` already exists elsewhere.

Reply:
Can't see a good place, so created `support/annex_utils.py` (we already have `archive_utils_*.py`)